src/OpenSpaceToolkit/Astrodynamics/Flight/System/SatelliteDynamics.cpp

```diff
@@ -0,0 +1,348 @@
+{
+
+    displayDecorator ? ostk::core::utils::Print::Header(anOutputStream, "SatelliteDynamics") : void () ;
+    ostk::core::utils::Print::Line(anOutputStream) << "State vector type:"                << SatelliteDynamics::StringFromStateVectorType(stateVectorType_) ;
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Environment:" << environment_ ;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Satellite System") ;
+    satelliteSystem_.print(anOutputStream, false) ;
+
+    ostk::core::utils::Print::Separator(anOutputStream, "State") ;
+    state_.print(anOutputStream, false) ;
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void () ;
+
+}
+
+SatelliteDynamics::StateVectorType       SatelliteDynamics::getStateVectorType                            ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("SatelliteDynamics") ;
+    }
+
+    return stateVectorType_ ;
+
+}
+
+State                           SatelliteDynamics::getState                 ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Satellite Dynamics") ;
+    }
+
+    return state_ ;
+
+}
+
+void                            SatelliteDynamics::setState                 (   const   State&                      aState                                      )
+{
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Satellite Dynamics") ;
+    }
+
+    state_ = aState ;
+    environment_.setInstant(aState.getInstant()) ;
+
+}
+
+SatelliteDynamics::DynamicalEquationFuncCallback  SatelliteDynamics::accessDynamicalEquations             ( ) const
+{
+
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("SatelliteDynamics") ;
+    }
+
+    return std::bind(&SatelliteDynamics::GravityTable_Dynamics, this, std::placeholders::_1,  std::placeholders::_2,  std::placeholders::_3) ;
+
+}
+
+String                          SatelliteDynamics::StringFromStateVectorType                                    (   const   SatelliteDynamics::StateVectorType&  aStateVectorType      )
+{
+
+    switch (aStateVectorType)
+    {
+
+        case SatelliteDynamics::StateVectorType::PositionVelocity:
+            return "PositionVelocity" ;
+
+        case SatelliteDynamics::StateVectorType::PositionVelocitywithDragCoefficient:
+            return "PositionVelocitywithDragCoefficient" ;
+
+        default:
+            throw ostk::core::error::runtime::Wrong("State Vector type") ;
+
+    }
+
+    return String::Empty() ;
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void                            SatelliteDynamics::GravityTable_Dynamics    (   const   SatelliteDynamics::StateVector&     x,
```

Comment:
Yes I agree, I need to rename this dynamics function and make it more general to handle perturbations from multiple bodies. However, it should be done similar to my commented out third body dynamics equation below with a for loop and not logicals since that will slow down performance.